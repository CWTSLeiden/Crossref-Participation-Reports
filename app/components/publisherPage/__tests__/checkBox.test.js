import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon'


import CheckBox from '../components/checkBox'



//Dummy Data




//Mocks
const sandbox = sinon.sandbox.create()
const didMount = sandbox.spy(CheckBox.prototype, 'componentDidMount')
const setState = sandbox.spy(CheckBox.prototype, 'setState')






//Render and begin tests
const mountComponent = mount(<CheckBox/>)


test('mount snapshot', () => {
  expect(didMount.callCount).toBe(1)
  expect(toJson(mountComponent)).toMatchSnapshot()
})


describe('componentDidMount', () => {
  test('should update tooltip positioning and add event listener', () => {

  })
})


describe('desktop or wide tablet', () => {
  test('if component on left side, tooltip on right', () => {

  })

  test('if component on right side, tooltip on left', () => {

  })

  test('resizing window small enough switches tooltip to left', () => {

  })

  test('resizing window big enough, switches tooltip to right', () => {

  })
})


describe('on mobile', () => {
  test('tooltip has special mobile styling', () => {

  })

  test('tooltip visible on icon click', () => {

  })

  test('tooltip closes on icon click or close button', () => {

  })
})


describe('componentWillUnmount', () => {
  test('should remove event listener', () => {

  })
})