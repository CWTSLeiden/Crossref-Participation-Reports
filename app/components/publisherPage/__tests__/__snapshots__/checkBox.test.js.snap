// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount snapshot 1`] = `
<CheckBox
  item={
    Object {
      "info": "tooltip text",
      "name": "Abstracts",
      "percentage": 39,
    }
  }
  setOpenTooltip={[Function]}
>
  <div
    className="check"
  >
    <div
      className="title"
    >
      Abstracts
      <div
        className="tooltipIconContainer"
      >
        <div
          className="hoverArea"
        >
          <img
            className="icon"
            src="/participationreports//images/Asset_Icons_Grey_Help.png"
            style={
              Object {
                "height": "22px",
                "width": "22px",
              }
            }
          />
          <img
            className="hoverIcon"
            onClick={null}
            src="/participationreports//images/Asset_Icons_LightGrey_Help.svg"
            style={
              Object {
                "display": "none",
                "height": "22px",
                "width": "22px",
              }
            }
          />
          <div
            className="tooltipHoverArea "
          />
          <div
            className="tooltipContentContainer   "
          >
            <p>
              tooltip text
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="barContainer"
    >
      <div
        className="bar"
      >
        <div
          className="barWidth"
          style={
            Object {
              "width": "39%",
            }
          }
        >
          <div
            className="animateBounce"
            style={
              Object {
                "backgroundColor": "#3EB1CB",
              }
            }
          />
        </div>
      </div>
      <div
        className="percent"
      >
        39
        <span>
          %
        </span>
      </div>
    </div>
  </div>
</CheckBox>
`;
